core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,20 +221,58 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          for (var i = 0; i < currentChangeList.length; i++) {
+            var changeDict =
+              currentChangeList[i].getBackendChangeObject();
+            if (changeDict.cmd === CMD_MOVE_SKILL_ID_TO_SUBTOPIC) {
+              if (changeDict.new_subtopic_id === subtopicId) {
+                if (changeDict.old_subtopic_id === null) {
+                  indicesToDelete.push(i);
+                } else {
+                  changeDict.cmd = CMD_REMOVE_SKILL_ID_FROM_SUBTOPIC;
+                  changeDict.subtopic_id = changeDict.old_subtopic_id;
+                  delete changeDict.old_subtopic_id;
+                  delete changeDict.new_subtopic_id;
+                }
+              } else if (changeDict.old_subtopic_id === subtopicId) {
+                changeDict.old_subtopic_id = null;
+              }
+            } else if (changeDict.cmd === CMD_REMOVE_SKILL_ID_FROM_SUBTOPIC) {
+              if (changeDict.subtopic_id === subtopicId) {
+                indicesToDelete.push(i);
+              }
+            }
+            currentChangeList[i].setBackendChangeObject(changeDict);
+          }
           for (var i = 0; i < currentChangeList.length; i++) {
             var backendChangeDict =
               currentChangeList[i].getBackendChangeObject();
-            if (backendChangeDict.subtopic_id === subtopicId) {
-              // The indices in the change list corresponding to changes to the
-              // currently deleted and newly created subtopic are to be removed
-              // from the list.
-              indicesToDelete.push(i);
+            if (indicesToDelete.indexOf(i) !== -1) {
```

Comment:
Might as well move this to line 248? Terminate as soon as possible.

Reply:
Done, removed this check as it was not necessary.